### PR TITLE
HSEARCH-4212 Official, documented compatibility with OpenSearch

### PR DESCRIPTION
**Pull request by yrodiere**
https://hibernate.atlassian.net/browse/HSEARCH-4212

### CODE REVIEW

**Comment by yrodiere** [comment]
Full build: https://ci.hibernate.org/blue/organizations/jenkins/hibernate-search/detail/PR-2602/2/pipeline

File: backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/ElasticsearchVersion.java [hunk 1]
```diff
@@ -76,6 +114,14 @@ public String toString() {
 		return builder.toString();
 	}
 
+	/**
+	 * @return The distribution to which  this version applies, e.g. {@link ElasticsearchDistributionName#ELASTIC}
```
Comment (fax4ever) [inline backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/ElasticsearchVersion.java:136]: there is a double spaces after `to which`

**Review by fax4ever — APPROVED** [review]
Nice job!
Just a comment

**Comment by fax4ever** [comment]
I think that we can rely on the [full build](https://ci.hibernate.org/blue/organizations/jenkins/hibernate-search/detail/PR-2602/4/pipeline/).

**Comment by yrodiere** [comment]
Addressed your comment and merged. Thanks!

**Pull request merged**